docusaurus/docs/Android/04-compose/07-guides/06-implementing-own-capabilities.mdx

```diff
@@ -0,0 +1,233 @@
+
+## Integrating channel capabilities into the UI
+
+If you are using our own UI in combination with our `ViewModel`s, then own capabilities work out of the box without any
+additional effort from your side, otherwise you will have to pass the correct set of capabilities to individual components.
+
+Different components have different ways of passing own capabilities to them, so we’ll go through them individually.
+
+:::note
+As a safety precaution, by default an empty set is passed wherever `ownCapabilities: Set<String>` is required. This gives the user no capabilities. Passing a set containing all capabilities will give the user potentially dangerous capabilities such as the ability to delete any message.
+:::
+
+### MessageComposer
+
+```kotlin
+@Composable
+public fun MessageComposer(
+        messageComposerState: MessageComposerState,
+        onSendMessage: (String, List<Attachment>) -> Unit,
+        modifier: Modifier = Modifier,
+        onAttachmentsClick: () -> Unit = {},
+        onCommandsClick: () -> Unit = {},
+        onValueChange: (String) -> Unit = {},
+        onAttachmentRemoved: (Attachment) -> Unit = {},
+        onCancelAction: () -> Unit = {},
+        onMentionSelected: (User) -> Unit = {},
+        onCommandSelected: (Command) -> Unit = {},
+        onAlsoSendToChannelSelected: (Boolean) -> Unit = {},
+        headerContent: @Composable ColumnScope.(MessageComposerState) -> Unit = {
+            // Default content
+        },
+        footerContent: @Composable ColumnScope.(MessageComposerState) -> Unit = {
+            // Default content
+        },
+        mentionPopupContent: @Composable (List<User>) -> Unit = {
+            // Default content
+        },
+        commandPopupContent: @Composable (List<Command>) -> Unit = {
+            // Default content
+        },
+        integrations: @Composable RowScope.(MessageComposerState) -> Unit = {
+            // Default content, uses own capabilities to regulate integrations
+        },
+        label: @Composable (MessageComposerState) -> Unit = {
+            // Default content
+        },
+        input: @Composable RowScope.(MessageComposerState) -> Unit = {
+            // Default content
+        },
+        trailingContent: @Composable (MessageComposerState) -> Unit = {
+            // Default content, uses own capabilities to regulate the send button
+        },
+)
+```
+
+`MessageComposer` contains a parameter `messageComposerState: MessageComposerState`. This class holds all the necessary
+state to make `MessageComposer` work, including a set of own capabilities. These are used by the default components
+hosted in the `MessageComposer`.
+
+The file containing `MessageComposer` can be
+found [here](https://github.com/GetStream/stream-chat-android/blob/0f1f54e912baf7a7284d9a4d1435fb6743aa9c78/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/composer/MessageComposer.kt#L214)
```

Comment:
missing a full stop at the end

Reply:
Also this section feels unfinished - we show the message composer file and then just jump over to the MessageComposerState, without any transitions.

Reply:
Added a full stop and added a more organic transition.